web_monitoring/cli.py

```diff
@@ -0,0 +1,55 @@
+# Command Line Interface
+# All the functionality is implemented in this module. See scripts/ for
+# executables.
+from docopt import docopt
+import hashlib
+import io
+import lxml.html
+import requests
+from tqdm import tqdm
+from web_monitoring import internetarchive as ia
+from web_monitoring import db
+
+
+# These functions lump together library code into monolithic operations for the
+# CLI. They also print. To access this functionality programmatically, it is
+# better to use the underlying library code.
+
+
+def import_ia(url, agency, site):
+    print('obtaining versions list from Internet Archive...')
+    versions = ia.list_versions(url)
+    # Collect all the results and POST them as a unit (rather than streaming).
+    formatted_versions = []
+    for dt, uri in tqdm(versions, desc='formatting versions'):
+        res = requests.get(uri)
+        assert res.ok
+        version_hash = hashlib.sha256(res.content)
+        content_as_file = io.StringIO(res.content.decode(errors='ignore'))
```

Comment:
of course, b/c you're super smart. Now that I understand it, yep I'd say you're totally fine to ignore errors.

If you have the bandwidth, I'd do a check on the `Content-Type` header for the `encoding` prop, switch on that for decoding, with the default decoding being `ASCII` / `ISO-8859-1`. Doing this groundwork to determine encoding is well worth it IMHO, as _many_ places now serve `UTF-8`, but `ISO-8859-1` is still the sensible default given the history of the web. This approach would keep strange unrecognized utf-8 characters out of your extracted titles, and set you up for future parsing stuuuuuuff